### PR TITLE
[Model Monitoring] Add project-level enable/disable + per-project OTel opt-in

### PR DESCRIPTION
**Pull request by davesh0812**
### 📝 Description

Adds project-level model monitoring state to the project spec and a per-project OTel export opt-in. Builds on the operator-side OTel scaffolding from #9668 (telemetry config + headers helper + function-pod mount injection) to give users a declarative SDK surface for opting their project into OTel and reading the resulting state back from the project spec.

Implements **ML-12543** — *Add `enable_otlp` flag to all relevant SDK/API*.

---

#### Schema + SDK
- **`mlrun/common/schemas/project.py`** — new `ProjectMonitoringSpec` model with `enabled`, `otlp_enabled`, `stream_type` (`StreamTarget`), `tsdb_type` (`TSDBTarget`). Nested under both `ProjectSpec` and `ProjectSpecOut` with `pydantic.v1.Field(default_factory=ProjectMonitoringSpec)` so the field is **always populated** — callers never have to None-guard `project.spec.model_monitoring.<flag>`. Stored as part of the existing `projects._full_object` blob — no SQL schema change.
- **`mlrun/common/schemas/model_monitoring/constants.py`** — new `StreamTarget` enum (`V3IO`, `KAFKA`) parallel to the existing `TSDBTarget`.
- **`mlrun/projects/project.py`** — SDK-side `ProjectMonitoringSpec(ModelObj)` mirror with property/setter so `project.spec.model_monitoring.<field>` works and round-trips through `to_dict()` / `from_dict()`. Setter **coerces `None` to a fresh default instance**, matching the schema's never-None contract.
- **`MlrunProject.enable_model_monitoring(...)`** gains `otlp_enabled: bool = False`. After the API call, the SDK re-fetches the project and merges the enriched server-side spec onto the local object via the same `self._enrich()` flow `save()` uses — so the local `project.spec.model_monitoring` immediately reflects `enabled=True, otlp_enabled=<param>` instead of staying stale.
- **`MlrunProject.set_model_monitoring_credentials(...)`** — same `_enrich(db.get_project(self.name))` refresh after the API call, so `stream_type` / `tsdb_type` derived server-side from the registered profile classes land on the local spec.
- **`MlrunProject.disable_model_monitoring(...)`** — same `_enrich(db.get_project(self.name))` refresh after the API call, so the server-side reset (`enabled=False`, `otlp_enabled=False`) is reflected on the local spec without an extra round-trip.

#### DB plumbing
- **`mlrun/db/{base,httpdb,nopdb}.py`** — `otlp_enabled` added to the `enable_model_monitoring` signature; forwarded as a query param to the server.

#### Server-side
- **`server/py/services/api/api/endpoints/model_monitoring.py`** — `PUT /projects/{p}/model-monitoring/` accepts `otlp_enabled`.
- **`server/py/services/api/crud/model_monitoring/deployment.py`** — biggest changes:
  - `deploy_monitoring_functions(otlp_enabled=False)` — **fail-fast validation** that rejects `otlp_enabled=True` when `mlrun.mlconf.telemetry.enabled` is falsy or `mlrun.mlconf.telemetry.otlp_endpoint` is blank (`MLRunBadRequestError`). The check uses **bool truthy** (`if not mlrun.mlconf.telemetry.enabled:`) since the env-loader coerces the value to a Python `bool` based on the schema default in `mlrun/config.py`. A prior version compared against the string `"true"` which always fired against the real bool value. After deploy, persists `enabled=True, otlp_enabled=<param>` to `project.spec.model_monitoring`.
  - `disable_model_monitoring` — declaratively resets both flags to `False`.
  - `set_credentials` — derives `stream_type`/`tsdb_type` from the resolved datastore profiles and persists them onto the project spec.
  - New `_persist_model_monitoring_spec(...)` helper: partial-update patch with `PatchMode.additive` so each call only writes the fields it owns.
  - New `_resolve_stream_target` / `_resolve_tsdb_target` static helpers map profile class → enum value.

---

### 🔄 End-to-end User Flow

```
SDK call                                Server effect                        Project spec result (local)
─────────────────────────────────────   ──────────────────────────────────   ────────────────────────────────────
project.set_model_monitoring_           Validate profiles, store profile     stream_type, tsdb_type populated
  credentials(                          names as project secrets, derive     (via _enrich(db.get_project(...)))
    tsdb_profile_name="my-pg",          types from profile classes
    stream_profile_name="my-kafka")

project.enable_model_monitoring(        Validate mlconf.telemetry.{enabled,  enabled=True, otlp_enabled=True
    otlp_enabled=True)                  otlp_endpoint}; deploy controller +  (via _enrich(db.get_project(...)))
                                        writer + stream functions

project.set_model_monitoring_           Set runtime.spec.otlp_enabled=True   (function-level — separate from spec)
  function(                             on the function; server-side
    name=..., otlp_enabled=True)        injection mounts the OTLP headers
                                        secret onto the pod (PR #9668)

project.disable_model_monitoring(...)   Tear down MM functions               enabled=False, otlp_enabled=False
                                                                             (stream/tsdb types preserved)
```

Resolution chain when launching a monitoring function:

```
runtime.spec.otlp_enabled (function-level explicit)
  └ inherits from
project.spec.model_monitoring.otlp_enabled (project-level opt-in, this PR)
  └ gated by
mlconf.telemetry.otlp_endpoint + mlconf.telemetry.enabled (operator config, PR #9668)
```

---

### ✅ Checklist
- [x] I updated the documentation (if applicable) — docstrings on new methods/fields
- [x] I have tested the changes in this PR — 41 unit tests across schema, SDK, and server crud
- [x] I confirmed whether my changes are covered by system tests
  - [x] Manually verified the patched API on a lab system (vmdev86ig4): a fresh `PUT /projects/{p}/model-monitoring/?otlp_enabled=true` against the patched image succeeds (telemetry bool-truthy fix), and `project.spec.model_monitoring` reflects `enabled=True, otlp_enabled=True, stream_type=kafka, tsdb_type=postgresql` immediately after the call (no extra `get_project` round-trip needed in user code).
- [ ] If I introduced a deprecation:
  - [ ] N/A
- [ ] Please run [Smoke-tests](https://github.com/mlrun/mlrun/actions/workflows/smoke-tests.yml) workflow

---

### 🧪 Testing

Run locally:
```
pytest tests/common/test_schemas.py \
       tests/projects/test_project.py::TestModelMonitoring \
       server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py
```

| Suite | Count | What it covers |
|---|---|---|
| `tests/common/test_schemas.py` | 6 | `ProjectMonitoringSpec` defaults; dict round-trip; invalid-enum rejection (`stream_type` / `tsdb_type`); `ProjectSpec.model_monitoring` defaults to a populated `ProjectMonitoringSpec()` (never None); nested round-trip; **`ProjectSpecOut.model_monitoring` default-populated + legacy-payload deserialization** (parallel to `ProjectSpec`) |
| `tests/projects/test_project.py::TestModelMonitoring` | 6 | SDK `ProjectSpec.model_monitoring` accepts dict (verifies `_verify_dict` conversion); round-trip through `to_dict`/`from_dict`; **setter coerces `None` → default instance**; `enable_model_monitoring` forwards `otlp_enabled` to mocked DB; **`enable_model_monitoring` re-fetches and merges the enriched server spec via `_enrich`**; **`set_model_monitoring_credentials` does the same `_enrich` refresh**; **`disable_model_monitoring` does the same `_enrich` refresh** (reset enabled/otlp_enabled to False) |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestOtelEnableValidation` | 4 | Rejects on disabled telemetry; rejects on blank endpoint; **passes happy path when `mlconf.telemetry.enabled=True` (bool) + endpoint set** (regression guard for the original string-compare bug); no check when `otlp_enabled=False` |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestPersistModelMonitoringSpec` | 2 | Partial-patch behavior: writes only provided fields; no-op on empty call |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestResolveStreamTarget` | 4 | Profile-class → `StreamTarget` enum: `KafkaStream` → `KAFKA`, `KafkaSource` (subclass) → `KAFKA`, `V3io` → `V3IO`, other → `None` |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestResolveTSDBTarget` | 3 | Profile-class → `TSDBTarget` enum: `PostgreSQL` → `TimescaleDB`, `V3io` → `V3IO_TSDB`, other → `None` |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestSetCredentialsPersistsResolvedTypes` | 1 | `set_credentials` end-to-end (external deps mocked): asserts the final `_persist_model_monitoring_spec` call carries `stream_type=KAFKA, tsdb_type=TimescaleDB` derived from the registered profile classes |
| `server/py/services/api/tests/unit/crud/model_monitoring/test_deployment.py::TestDisableModelMonitoringPersistsDisabledSpec` | 1 | `disable_model_monitoring` declaratively writes `enabled=False, otlp_enabled=False` to the spec at the end of teardown |

**Total: 41 tests** (suite passes locally).

Manual lab verification (vmdev86ig4): patched the API server with this branch's image (`1.12.0-rc1`). The end-to-end flow from a notebook — `register_datastore_profile` → `set_model_monitoring_credentials` → `enable_model_monitoring(otlp_enabled=True)` — produces a local `project.spec.model_monitoring` that matches the server-side state without extra round-trips.

---

### 🔗 References

- **Ticket**: [ML-12543 – Add `enable_otlp` flag to all relevant SDK/API](https://ecliptos.atlassian.net/browse/ML-12543)
- **Parent epic**: [ML-12344 – Model monitoring OpenTelemetry support](https://ecliptos.atlassian.net/browse/ML-12344)
- **Builds on**: #9668 (telemetry config + headers helper + function-pod mount injection)

---

### 🚨 Breaking Changes?

- [ ] Yes (explain below)
- [x] No

---

### CODE REVIEW

File: server/py/services/api/crud/model_monitoring/deployment.py [hunk 1]
```diff
@@ -2133,6 +2227,11 @@ def set_credentials(
             ),
         )
 
+        self._persist_model_monitoring_spec(
+            stream_type=self._resolve_stream_target(stream_profile),
```
Comment (tomerm-iguazio) [inline server/py/services/api/crud/model_monitoring/deployment.py:2234]: 🔴 **Bug: potential `NameError` when `stream_profile_name` is `None`**

`stream_profile` is only assigned inside `if stream_profile_name:`, so if that branch is skipped, `stream_profile` is undefined when this line executes — raising `NameError` at runtime.

Fix: initialize `stream_profile = None` before the `if` block. `_resolve_stream_target(None)` already returns `None` gracefully (all `isinstance` checks fail), and `_persist_model_monitoring_spec` skips `None` values — so a one-line init is sufficient.

Also missing a test for the `set_credentials(stream_profile_name=None)` path.

Comment (tomerm-iguazio) [reply server/py/services/api/crud/model_monitoring/deployment.py:2234]: Same issue existed in dev - under `set_credentials`:
```
if tsdb_profile_name:
    tsdb_profile = self._validate_and_get_tsdb_profile(tsdb_profile_name)
    secrets_dict[
        mlrun.common.schemas.model_monitoring.ProjectSecretKeys.TSDB_PROFILE_NAME
    ] = tsdb_profile_name
....
self._persist_model_monitoring_spec(
    stream_type=self._resolve_stream_target(stream_profile),
    tsdb_type=self._resolve_tsdb_target(tsdb_profile),
)
```
please set `tsdb_profile=None` as well in the beginning of the function.

**Review by tomerm-iguazio — CHANGES_REQUESTED** [review]
Nice work! Few comments:
 ---

**🟡 Extra round-trip per mutation**

`enable_model_monitoring`, `disable_model_monitoring`, and `set_model_monitoring_credentials` each add a `db.get_project()` call to refresh local state. Worth a follow-up: returning the updated `ProjectMonitoringSpec` in the API response body would eliminate the extra HTTP call per operation.

File: server/py/services/api/crud/model_monitoring/deployment.py (hunk 1 above)
Comment (davesh0812) [reply server/py/services/api/crud/model_monitoring/deployment.py:2234]: Good catch, thanks. Applied the defensive fix — stream_profile = None / tsdb_profile = None at the top of set_credentials.
  
  Worth noting: the path isn't actually reachable today — the mandatory_secrets() check ~20 lines above the _resolve_*_target calls raises MLRunInvalidMMStoreTypeError whenever either profile name is missing, so neither
  variable is ever None when those calls run. But you're right that static analysers can't see that, and a future refactor of the mandatory-secrets contract would turn this into a real NameError. Defensive init costs nothing —
  taking it.

**Review by tomerm-iguazio — APPROVED** [review]
LGTM:)
